### ISSUE TITLE
Upload failure does not explain at all why upload failed in case of login failure

### ISSUE THREAD

**Issue by matkoniecz** [commons-app/apps-android-commons#4707]
**Summary:** 

On upload failure I just get "Failed" with no explanation what happened.

Is my account blocked? No.

Is there is no internet connection? There is.

Maybe app lost ability to open files? ????

Maybe there was an API upgrade? ????

Maybe this app version was remotely disabled? ????

Maybe there is something wrong with this file? ????

Maybe there is something wrong with file name? Duplicated? No. Using forbidden characters? No, happens also with pure ASCII

Several attempts later and after filing this issue: wait, have I changed password? Yes.

**Steps to reproduce:** 

Have working install.

Change password (outside app, at https://commons.wikimedia.org)

Forget about it.

Try to upload file.

Become confused why it is impossible as getting "Failed" as sole message is not helpful at all.

**System logs:**

I have trouble with connecting phone by cable to laptop - the port is damaged. It is still charging but getting data connection is nearly impossible.

But of error specific messages are existing already and I run into catchall for mysterious issues I will try to connect.

But it takes about 30 minutes.

```
Add logcat files here (if possible).

Need help? See https://github.com/commons-app/apps-android-commons/wiki/Getting-app-logs-from-Android-Studio
```

**Device and Android version:** 

Mi A2 Lite (reminder to myself, do not buy Xiaomi just because it is cheaper - it falls almost immediately, just after two years and touch screen starts tapping itself on its own)

Android 10, update May 2020 (and system updates stop quickly)

stock version from the manufacturer
 
**Commons app version:** 

3.1.1~1c9267ca0

**Screen-shots:** 

Can be created by pressing the Volume Down and Power Button at the same time on Android 4.0 and higher.

**Would you like to work on the issue?**

No, I sadly have not enough time for more tasks.

PS: Logging out is flaky, I tried to log out multiple times before I succeeded.

**Comment by misaochan** [comment]
Hi @matkoniecz , sorry to hear that! Unfortunately, as far as stating the reason, we are limited to what the server tells us (which is not much, and we have no control over the server, as we are a community team and not employed by WMF).

> Several attempts later and after filing this issue: wait, have I changed password? Yes.

Sorry, I just want to clarify this. So after you logged out and logged back in with your changed password, the uploads work?

**Comment by matkoniecz** [comment]
> Just want to clarify this: so after you logged out and logged back in with your changed password, the uploads work?

Yes

> Unfortunately, as far as stating the reason, we are limited to what the server tells us (which is not much

Even "upload rejected by Wikimedia Commons server, reason unknown" would be already an improvement! 

I suspected that app failed to open file or that internet failed.

Info that upload was refused by server would already help a lot in this case.

> which is not much, and we have no control over the server

If it is not making a specific error about authentication failure then I would be happy to make a feature request about this problem

**Comment by misaochan** [comment]
> Info that upload was refused by server would already help a lot in this case.

Makes sense! Yes, this should be doable I believe.

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul i would like to work on this issue please assign me

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul  I was working on this issue and this seems non-reproducible to me, like how to change the password there is no such option in the app

**Comment by sivaraam** [comment]
This is a duplicate of #5522 which has been fixed in v5.0.1. So, closing this. 🙂
